src/streamlink/plugins/twitch.py

```diff
@@ -254,19 +257,21 @@ def __init__(self, session, beta=False, version=3):
     def add_cookies(self, cookies):
         self.session.http.parse_cookies(cookies, domain="twitch.tv")
 
-    def call(self, path, format="json", schema=None, **extra_params):
+    def call(self, path, format="json", schema=None, private=False, **extra_params):
         params = dict(as3="t", **extra_params)
 
-        if self.oauth_token:
-            params["oauth_token"] = self.oauth_token
-
         if len(format) > 0:
             url = "https://{0}.twitch.tv{1}.{2}".format(self.subdomain, path, format)
         else:
             url = "https://{0}.twitch.tv{1}".format(self.subdomain, path)
 
         headers = {'Accept': 'application/vnd.twitchtv.v{0}+json'.format(self.version),
-                   'Client-ID': TWITCH_CLIENT_ID}
+                   'Client-ID': TWITCH_CLIENT_ID if not private else TWITCH_CLIENT_ID_PRIVATE}
+
+        # OAuth tokens created from Streamlink's own client-id can't be used anymore on the private API (#2680)
+        # Since we don't know the origin of the provided OAuth token, we unfortunately need to disable all
```

Comment:
Regarding what I said here, it \*is\* possible to check whether it's an OAuth token from Twitch's website or not. That would mean that we don't have to disable the `Authorization` header as a whole on the private API endpoints and people could still log in via an OAuth token from Twitch's website. We would have to make another API request for the auth permissions on the kraken root endpoint first though, which would delay launching streams by a bit when an OAuth token is provided. This does only work because certain permission scopes can only be granted by Twitch and thus are not available on 3rd party OAuth tokens, but I'm not sure if that's a proper and solid way, as those permissions can and probably will change over time. Thoughts?

Reply:
> but I'm not sure if that's a proper and solid way, as those permissions can and probably will change over time. Thoughts?

don't think to complicated, working Streams should be enough for now.

---

yt-dl also uses the api key since 8-2018

https://github.com/ytdl-org/youtube-dl/commit/cb1c3a3c07b2afe854ecd8c97f77d8f62720d7a4